stdlib/logging/config.pyi

```diff
@@ -19,7 +21,29 @@ DEFAULT_LOGGING_CONFIG_PORT: int
 RESET_ERROR: int  # undocumented
 IDENTIFIER: Pattern[str]  # undocumented
 
-def dictConfig(config: dict[str, Any]) -> None: ...
+class _RootLoggerConfiguration(TypedDict, total=False):
+    level: _Level
+    filters: Sequence[str]
+    handlers: Sequence[str]
+
+class _LoggerConfiguration(_RootLoggerConfiguration, TypedDict, total=False):
+    propagate: bool
+
+class _OptionalDictConfigArgs(TypedDict, total=False):
+    # these two can have custom factories (key: `()`) which can have extra keys
+    formatters: dict[str, dict[str, Any]]
+    filters: dict[str, dict[str, Any]]
+    # type checkers would warn about extra keys if this was a TypedDict
+    handlers: dict[str, dict[str, Any]]
+    loggers: dict[str, _LoggerConfiguration]
+    root: _RootLoggerConfiguration | None
```

Comment:
coming to this late -- should this be `NotRequired` instead of `| None`?  running into some issues with this

Reply:
Well, it is `NotRequired` by virtue of `_OptionalDictConfigArgs`  using `total=False`. (This code probably predates PEP 655 and could be updated.) The implementation uses `config.get("root")` to access this item:

https://github.com/python/cpython/blob/48dfd74a9db9d4aa9c6f23b4a67b461e5d977173/Lib/logging/config.py#L540C28-L540C28

That said, I think allowing `None` here is incidental, so if this makes problems, I'd be okay with removing it.

Reply:
will send a PR!  basically having issues with `x: _DictConfigArgs` + `x["root"]["level"] = ...`

Reply:
#11077 for that! we'll see how mypy-primer reacts